File: EESoft/DataCenter/DataCenter-Network Service/OptionsFrm.cs
using System.Windows.Forms;
using System;
using Microsoft.Win32;

namespace DataCenterNetworkService
{
	public partial class OptionsFrm
	{
		public OptionsFrm()
		{
			InitializeComponent();
		}
		private enum RegAutoStartKeyValues
		{
			GeneralPort,
			DeletedFilesPath,
			DataCenterAutoStart
		}
        public void OptionsFrm_Load(Object sender, EventArgs e)
        {
            try
            {
                string temp_string = "";
                if (GetRegAutoStartValue(ref temp_string, RegAutoStartKeyValues.DataCenterAutoStart) == true)
                    CBAutoStart.Checked = true;
                else
                    CBAutoStart.Checked = false;
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
                CBAutoStart.Enabled = false;
                BtnApply.Enabled = false;
            }
        }
        public void BtnApply_Click(Object sender, EventArgs e)
        {
            try
            {
                if (CBAutoStart.Checked == true)
                    SetRegAutoStartValue(Application.ExecutablePath, RegAutoStartKeyValues.DataCenterAutoStart);
                else
                    DelRegAutoStartValue(RegAutoStartKeyValues.DataCenterAutoStart);
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }
		private bool GetRegAutoStartValue(ref string KeyValue, RegAutoStartKeyValues ValueName)
		{
			//MModule.PrepareReg();
			string[] arr = Registry.CurrentUser.OpenSubKey("SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Run", false).GetValueNames();
			int ForInt = 0;
			for (ForInt = 0; ForInt <= arr.Length - 1; ForInt++)
			{
				if (arr[ForInt] == ValueName.ToString())
				{
					KeyValue = Registry.LocalMachine.OpenSubKey("SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Run", false).GetValue(ValueName.ToString()).ToString();
					return true;
				}
			}
			return false;
		}
		private void SetRegAutoStartValue(string KeyValue, RegAutoStartKeyValues ValueName)
		{
			//MModule.PrepareReg();
			Registry.CurrentUser.OpenSubKey("SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Run", true).SetValue(ValueName.ToString(), KeyValue);
		}
		private void DelRegAutoStartValue(RegAutoStartKeyValues ValueName)
		{
			//MModule.PrepareReg();
			Registry.CurrentUser.OpenSubKey("SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Run", true).DeleteValue(ValueName.ToString());
		}
	}
}

File: EESoft/DataCenterVB/DataCenter-Network Service/OptionsFrm.vb
Imports Microsoft.Win32
Public Class OptionsFrm
    Private Enum RegAutoStartKeyValues
        GeneralPort
        DeletedFilesPath
        DataCenterAutoStart
    End Enum
    Private Sub OptionsFrm_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Try
            If GetRegAutoStartValue("", RegAutoStartKeyValues.DataCenterAutoStart) = True Then
                CBAutoStart.Checked = True
            Else
                CBAutoStart.Checked = False
            End If
        Catch ex As Exception
            MsgBox(ex.Message)
            CBAutoStart.Enabled = False
            BtnApply.Enabled = False
        End Try
    End Sub
    Private Sub BtnApply_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BtnApply.Click
        Try
            If CBAutoStart.Checked = True Then
                SetRegAutoStartValue(Application.ExecutablePath, RegAutoStartKeyValues.DataCenterAutoStart)
            Else
                DelRegAutoStartValue(RegAutoStartKeyValues.DataCenterAutoStart)
            End If
        Catch ex As Exception
            MsgBox(ex.Message)
        End Try
    End Sub
    Private Function GetRegAutoStartValue(ByRef KeyValue As String, ByVal ValueName As RegAutoStartKeyValues) As Boolean
        PrepareReg()
        Dim arr() As String = Registry.CurrentUser.OpenSubKey("SOFTWARE\Microsoft\Windows\CurrentVersion\Run", False).GetValueNames
        Dim ForInt As Integer = 0
        For ForInt = 0 To arr.Length - 1
            If arr(ForInt) = ValueName.ToString Then
                KeyValue = Registry.LocalMachine.OpenSubKey("SOFTWARE\Microsoft\Windows\CurrentVersion\Run", False).GetValue(ValueName.ToString)
                Return True
            End If
        Next
        Return False
    End Function
    Private Sub SetRegAutoStartValue(ByVal KeyValue As String, ByVal ValueName As RegAutoStartKeyValues)
        PrepareReg()
        Registry.CurrentUser.OpenSubKey("SOFTWARE\Microsoft\Windows\CurrentVersion\Run", True).SetValue(ValueName.ToString, KeyValue)
    End Sub
    Private Sub DelRegAutoStartValue(ByVal ValueName As RegAutoStartKeyValues)
        PrepareReg()
        Registry.CurrentUser.OpenSubKey("SOFTWARE\Microsoft\Windows\CurrentVersion\Run", True).DeleteValue(ValueName.ToString)
    End Sub
End Class